sim-lib/src/lib.rs

```diff
@@ -219,6 +219,8 @@ enum SimulationEvent {
 
 /// SimulationOutput provides the output of a simulation event.
 enum SimulationOutput {
+    /// Terminal error to signal simulation exit
+    TerminalError(SimulationError),
```

Comment:
Implication:
- any task or method call should send this `SimulationOutput` to cleanly terminate the simulation
- in case sending this specific output to the channel is not possible, `shutdown` trigger should be used, to the same effect but without the final log of why the simulation terminated.